### ISSUE TITLE
Unexpected error diagnostics in Typescript 5.2.2

### ISSUE THREAD

**Issue by lramos15** [microsoft/TypeScript#56266]
Working on https://github.com/microsoft/vscode-extension-telemetry updating all packages to latest (including typescript 5.2.2) appears to have caused unexpected red underlining telling me I don't have the node types when I have them installed. 

Attempting to install the missing types via the quick fix leads to a false expression error notification
![image](https://github.com/microsoft/vscode/assets/4544166/8bb41533-aab2-41ec-ab57-758d9498b3cf)

**Comment by RyanCavanaugh** [comment]
What I did:
 * git clone the repo
 * npm install typescript@5.2
 * code .
 * Open src\node\telemetryReporter.ts
 * No errors
 * 🤷‍♂️

Repro steps please?

**Comment by lramos15** [comment]
Remove `"applicationinsights": "^2.7.1"` from the package JSON and `npm install`. That seems to be what triggers it for me

**Comment by RyanCavanaugh** [comment]
> but not an error regarding the built-in node packages.

Regarding the error -- you explicitly have type directive searching disabled due to `typeRoots: []` in the tsconfig. I'm guessing `applicationinsights` got it brought in in a roundabout way, but TS is doing what you told it to.
```ts
"typeRoots": []
```

**Comment by RyanCavanaugh** [comment]
@sheetalkamat here's the error trace. `@types/node` is already in package.json under devdeps
```
Err 280   [15:47:21.799] Exception on executing command {
  "seq": 65,
  "type": "request",
  "command": "applyCodeActionCommand",
  "arguments": {
    "command": {
      "type": "install package",
      "file": "d:/Throwaway/vscode-extension-telemetry/src/node/telemetryReporter.ts",
      "packageName": "@types/node"
    }
  }
}:

    Debug Failure. False expression.

    Error: Debug Failure. False expression.
        at _NodeTypingsInstaller.installPackage (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:186191:13)
        at TypingsCache.installPackage (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:175906:27)
        at ConfiguredProject2.installPackage (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:176180:30)
        at applySingleCodeActionCommand (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144424:39)
        at Object.applyCodeActionCommand (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144419:96)
        at IpcIOSession.applyCodeActionCommand (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:184567:36)
        at applyCodeActionCommand (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:182680:43)
        at c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:184838:69
        at IpcIOSession.executeWithRequestId (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:184830:14)
        at IpcIOSession.executeCommand (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:184838:29)
        at IpcIOSession.onMessage (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:184880:51)
        at process.<anonymous> (c:\Users\ryanca\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:186461:14)
        at process.emit (node:events:513:28)
        at emit (node:internal/child_process:937:14)
        at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
Info 281  [15:47:21.799] response:
    {"seq":0,"type":"response","command":"applyCodeActionCommand","request_seq":65,"success":false,"message":"Error processing request. Debug Failure. False expression.\nError: Debug Failure. False expression.\n    at _NodeTypingsInstaller.installPackage (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:186191:13)\n    at TypingsCache.installPackage (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:175906:27)\n    at ConfiguredProject2.installPackage (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:176180:30)\n    at applySingleCodeActionCommand (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:144424:39)\n    at Object.applyCodeActionCommand (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:144419:96)\n    at IpcIOSession.applyCodeActionCommand (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:184567:36)\n    at applyCodeActionCommand (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:182680:43)\n    at c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:184838:69\n    at IpcIOSession.executeWithRequestId (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:184830:14)\n    at IpcIOSession.executeCommand (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:184838:29)\n    at IpcIOSession.onMessage (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:184880:51)\n    at process.<anonymous> (c:\\Users\\ryanca\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\extensions\\node_modules\\typescript\\lib\\tsserver.js:186461:14)\n    at process.emit (node:events:513:28)\n    at emit (node:internal/child_process:937:14)\n    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"}
```

**Comment by lramos15** [comment]
> > but not an error regarding the built-in node packages.
> 
> Regarding the error -- you explicitly have type directive searching disabled due to `typeRoots: []` in the tsconfig. I'm guessing `applicationinsights` got it brought in in a roundabout way, but TS is doing what you told it to.
> 
> ```ts
> "typeRoots": []
> ```

Thanks for the fix! Didn't realize I had those
